src/python/zquantum/core/wip/circuits/_gates.py

```diff
@@ -13,6 +13,14 @@
 
 Parameter = Union[sympy.Symbol, Number]
 
+def _get_free_symbols(parameters: Tuple[Parameter, ...]) -> Iterable[sympy.Symbol]:
```

Comment:
Not sure about what this refactor actually does. Could you elaborate?

Reply:
So originally, `MatrixFactoryGate` was borrowing the `free_symbols` method from `Gate`: https://github.com/zapatacomputing/z-quantum-core/blob/6f034fa0353e2307cc2e0cfe55e8d343d777b51d/src/python/zquantum/core/wip/circuits/_gates.py#L247-L250

However if you tried to call the `free_symbols` property of a `MatrixFactoryGate`, mypy would complain because `self` argument had the wrong type. So I just put the logic for getting the symbols into the function `_get_free_symbols`, and refactored the methods of `Gate` and `MatrixFactoryGate` to call that function.

Reply:
👍